snooty/postprocess.py

```diff
@@ -1141,6 +1141,39 @@ def enter_node(self, fileid_stack: FileIdStack, node: n.Node) -> None:
             return
 
 
+class InstruqtHandler(Handler):
+    """Identify if Instruqt directive is present on a page and add title as a page-level option if so"""
+
+    def __init__(self, context: Context) -> None:
+        super().__init__(context)
+        self.has_instruqt_directive = False
+        self.instruqt_title = ""
+
+    def enter_page(self, fileid_stack: FileIdStack, page: Page) -> None:
+        self.has_instruqt_directive = False
+        self.instruqt_title = ""
+
+    def exit_page(self, fileid_stack: FileIdStack, page: Page) -> None:
+        if not self.has_instruqt_directive:
+            return
+
+        else:
+            page.ast.options["title"] = [self.instruqt_title]
```

Comment:
I would rather the option name be e.g. `instruqt_title` rather than just `title`. And I'd prefer not to have the list wrapper around the value